authentication/views.py

```diff
@@ -439,3 +442,293 @@ def checkAccountExists(request):
       username = params['username']
       result['usernameExist'] = Credential.objects.all().filter(username=username).filter(partnerId=partnerId).exists()
     return HttpResponse(json.dumps(result), status=status.HTTP_200_OK);
+
+#/credentials/checkOrcid
+class CheckOrcid(generics.GenericAPIView):
+    requireApiKey = False
+    queryset = Credential.objects.all()
+    
+    def get_queryset(self):
+        user_identifier = self.request.query_params.get('userIdentifier')
+        partner_id = self.request.query_params.get('partnerId')
+        return self.queryset.filter(userIdentifier=user_identifier, partnerId__partnerId=partner_id)
+
+    def get(self, request, *args, **kwargs):
+        user_identifier = request.query_params.get('userIdentifier')
+        partner_id = request.query_params.get('partnerId')
+
+        if not user_identifier or not partner_id:
+            return Response({'error': 'Both userIdentifier and partnerId are required.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if partner_id.lower() != 'tair':
+            return Response({'error': 'This check is only available for TAIR users.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            credential = self.get_queryset().get()
+        except Credential.DoesNotExist:
+            return Response({'error': 'User not found.'}, status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            orcid_credential = OrcidCredentials.objects.get(
+                credential=credential,
+                orcid_id__isnull=False
+            )
+            has_orcid = True
+            orcid_id = orcid_credential.orcid_id
+        except OrcidCredentials.DoesNotExist:
+            has_orcid = False
+            orcid_id = None
+
+        return Response({
+            'has_orcid': has_orcid,
+            'orcid_id': orcid_id
+        })
+
+checkOrcid = CheckOrcid.as_view()
+    
+#/credentials/getUserIdentifierByOrcid
+class GetUserIdentifierByOrcid(generics.GenericAPIView):
+    requireApiKey = False
+    queryset = OrcidCredentials.objects.all()
+
+    def get_queryset(self):
+        orcid_id = self.request.query_params.get('orcidId')
+        return self.queryset.filter(orcid_id=orcid_id)
+
+    def get(self, request, *args, **kwargs):
+        orcid_id = request.query_params.get('orcidId')
+        
+        if not orcid_id:
+            return Response({'error': 'ORCID ID is required.'}, status=status.HTTP_400_BAD_REQUEST)
+        
+        try:
+            orcid_credential = self.get_queryset().get()
+            user_identifier = orcid_credential.credential.userIdentifier
+            return Response({'userIdentifier': user_identifier})
+        except OrcidCredentials.DoesNotExist:
+            return Response({'userIdentifier': None})
+
+getUserIdentifierByOrcid = GetUserIdentifierByOrcid.as_view()
+
+
+#/credentials/addOrcidCredentials
+class AddOrcidCredentials(generics.GenericAPIView):
```

Comment:
I am a bit confused. Any authentication in this process? Does that mean anyone can call this API and update ORCID values for any user identifier?

Reply:
Solution: use credentialId and secretKey to authenticate the user. Only change ORCID info for that user. No userIdentifier needed.